WomanDayBot/Dialogs/Greeting/GreetingDialog.cs

```diff
@@ -84,8 +84,8 @@ private async Task<DialogTurnResult> AcknowledgeUserDataAsync(WaterfallStepConte
             // Return the collected information to the parent context.
             var userData = new UserData
             {
-                Name = Name,
-                Room = Room
+                Name = (string)stepContext.Values[Name],
```

Comment:
To prevent InvalidCastException would be better to use `as` operator. Example: `stepContext.Values[Name] as string`